File: Application/src/main/java/com/shareyourproxy/api/rx/command/eventcallback/UserMessageAddedEventCallback.java
package com.shareyourproxy.api.rx.command.eventcallback;

import android.os.Parcel;
import android.support.annotation.NonNull;

import com.shareyourproxy.api.domain.model.Message;

import java.util.HashMap;

/**
 * Created by Evan on 6/18/15.
 */
public class UserMessageAddedEventCallback extends EventCallback {
    private final static java.lang.ClassLoader CL =
        UserMessageAddedEventCallback.class.getClassLoader();
    public static final Creator<UserMessageAddedEventCallback> CREATOR =
        new Creator<UserMessageAddedEventCallback>() {
            @Override
            public UserMessageAddedEventCallback createFromParcel(Parcel in) {
                return new UserMessageAddedEventCallback(
                    (HashMap<String, Message>) in.readValue(CL));
            }

            @Override
            public UserMessageAddedEventCallback[] newArray(int size) {
                return new UserMessageAddedEventCallback[size];
            }
        };
    public final HashMap<String, Message> message;

    /**
     * Public constructor.
     *
     * @param message notification content
     */
    public UserMessageAddedEventCallback(@NonNull HashMap<String, Message> message) {
        this.message = message;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(message);
    }
}

File: Application/src/main/kotlin/com/shareyourproxy/api/rx/command/eventcallback/UserMessageAddedEventCallback.kt
package com.shareyourproxy.api.rx.command.eventcallback

import android.os.Parcel
import android.os.Parcelable
import com.shareyourproxy.api.domain.model.Message

/**
 * User message added.
 * @param message notification content
 */
internal final class UserMessageAddedEventCallback(val message: Message) : EventCallback() {

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(message)
    }

    companion object {
        private val CL = UserMessageAddedEventCallback::class.java.classLoader
        @JvmField val CREATOR: Parcelable.Creator<UserMessageAddedEventCallback> = object : Parcelable.Creator<UserMessageAddedEventCallback> {
            @Suppress("UNCHECKED_CAST")
            override fun createFromParcel(parcel: Parcel): UserMessageAddedEventCallback {
                return UserMessageAddedEventCallback(parcel.readValue(CL) as Message)
            }

            override fun newArray(size: Int): Array<UserMessageAddedEventCallback?> {
                return arrayOfNulls(size)
            }
        }
    }
}
